File: leetcode/go/27.remove-element.go
/*
 * @lc app=leetcode id=27 lang=golang
 *
 * [27] Remove Element
 */

// @lc code=start
func removeElement(nums []int, val int) int {
	length := len(nums)
	ret := 0
	for i:= 0;i < length; i++{
		if nums[i] != val {
			nums[ret] = nums[i]
			ret++
		}
	}
    return ret
}
// @lc code=end

File: leetcode/cpp/27.remove-element.cpp
/*
 * @lc app=leetcode id=27 lang=cpp
 *
 * [27] Remove Element
 */
#include <iostream>
#include <vector>
using namespace std;
// @lc code=start
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int pos = 0;
        int i = 0;
        for(; i < nums.size(); i++){
            if(nums[i] != val){
                if(pos != i){
                    nums[pos] = nums[i];   
                }
                ++pos;
            }
        }
        return pos;
    }
};
// @lc code=end

